test/functional/holds_controller_test.rb

```diff
@@ -32,12 +32,34 @@ class HoldsControllerTest < ActionController::TestCase
     assert_response :success
   end
 
-  test "should update hold successfully" do
-    post :update, params: { id: @hold1.access_key, hold_change: {"comment" => "qqq", "status" => "cancelled"}, hold: {status: "MyText"} }
+  test "update teacher-set available_copies and cancel hold successfully" do
+    sign_in @user
+    expected_ts_available_copies  = 4
```

Comment:
added functional test cases.  
Case1: Before cancellation of hold teacher-set available_copies count is 2. Once hold is cancelled,  status also will change to new to cancelled and based on the quantity we are updating the teacher-set available_copies count.

Case2: If hold is not cancelled  status also not changed.